wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -176,90 +179,206 @@ Example of an inherited Timing Point:
+### Common structure
 
-`x,y,time,type,hitSound,addition`
+This section is made of CSV lines. The first 5 fields are common to every hit objects, and an optional last *addition* fields.
 
-`164,260,2434,1,0,0:0:0:0:`
+**Syntax**: `x,y,time,type,hitSound...,addition`
 
-A hit circle is a single hit in all osu! game modes.
+#### Position
+
+*x* and *y* are integers representing the position of the center of the hit object. *x* ranges from 0 to 512 pixels, inclusive, and *y* ranges from 0 to 384 pixels, inclusive. The origin, (0, 0) is at the top left of the screen.
+
+To map these coordinates for a standard 640x480 screen, you need to add 64 pixels to *x* and 48 pixels to *y* to respect a uniform padding. Without the padding, an object at (0, 0) will be cut on the top left for the screen.
+
+Hit objects whose position is irrelevant, like spinners in osu! or any note in osu!taiko, have the special position (256,192), which is the center of the screen. For osu!mania, only *x* is relevant.
+
+#### Time
+
+*time* is an integral number of milliseconds from the beginning of the song, and specifies when the hit begins.
+
+#### Type
+
+*type* is a bitmap specifying the object type and attributes.
+
+- Bit 0 (1): circle.
+- Bit 1 (2): slider.
+- Bit 2 (4): new combo.
+- Bit 3 (8): spinner.
+- Bits 4-6 (16, 32, 64) form a 3-bit number (0-7) that chooses how many combo colors to skip. Note that 0 means 1 color is skipped anyway by the new combo field. A skip value of 1 therefore skips 2 colors.
+- Bit 7 (128) is for an osu!mania hold note.
+
+Circles, sliders, spinners, and hold notes can be OR'd with new combos and the combo skip value, but not with each other.
+
+Unconfirmed: a non-zero combo skip value is ignored when the new combo bit is not set.
+
+Examples:
+
+- `1`: circle.
+- `5 = 1 + 4`: circle starting a new combo.
+- `22 = 16 + 4 + 2`: slider starting a new combo, skipping 2 colors.
+
+#### Hit sounds
+
+*hitSound* (Integer) is a bitmap of hit sounds to play when the hit object is successfully hit.
+
+- Unconfirmed: Bit 0 (1): normal.
+- Bit 1 (2): whistle.
+- Bit 2 (4): finish.
+- Bit 3 (8): clap.
+
+Unconfirmed: The special value 0 is the same as 1 (hitnormal).
+
+Unconfirmed: If bit 0 is not set, the normal sound is not played.
+
+Unconfirmed: Multiple sounds will overlap if multiple bits are set.
+
+The sample set and custom index are usually specified in the timing point associated to the hit object, but may be customized in the *addition* field.
+
+The filename of the sample to play is `{sample set}-hit{sound}{index}.wav`, where:
+
+- *sample set* is normal, soft, or drum.
+- *sound* is normal, whistle, finish, or clap.
+- *index* is the custom index. It is omitted when equal to 0 or 1.
+
+#### Addition
+
+The *addition* field is optional and define extra parameters related to the hit sound samples. It defaults to `0:0:0:0:`.
+
+Unconfirmed: This field is deprecated. You may omit it entirely, or write `0:0:0:0:` instead.
```

Comment:
it's not deprecated at all, this is used in almost every ranked map

Reply:
So far I've only downloaded beatmap packs, and never found a single beatmap using that feature. I'll check out some ranked beatmaps to extract some examples.

Reply:
most hitobjects will be all zero, because they can simply inherit that information from timing points, but there should be at least a few in nearly every map that's properly hitsounded. [random example](https://osu.ppy.sh/b/1195081)